clients/unshaded/src/main/java/tachyon/client/FileOutStream.java

```diff
@@ -187,25 +181,23 @@ public void write(byte[] b, int off, int len) throws IOException {
         int tLen = len;
         int tOff = off;
         while (tLen > 0) {
-          if (mCurrentBlockLeftByte == 0) {
+          if (mCurrentBlockOutStream == null) {
+            getNextBlock();
+          } else if (mCurrentBlockOutStream.getRemainingSpaceByte() == 0) {
```

Comment:
if it is the first time to write data into the file, create a new block
if it gets the end of the block, create another new block